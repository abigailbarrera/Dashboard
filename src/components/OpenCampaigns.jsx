import React, { useContext } from "react";
import { GlobalContext } from "../context/GlobalState";
import { Link } from "react-router-dom";
import { Table, Container, Row } from 'react-bootstrap';


export const OpenCampaigns = () => {
    const { campaigns, editCampaign } = useContext(GlobalContext);
    const openCampaigns = campaigns.filter(campaign => campaign.status !== 'Sent');
    const {segments} = React.useContext(GlobalContext);

    return (
        <Container>
            <Row>
                <div className="col-12 mt-5">
                    <Table striped bordered hover>
                        <thead>
                            <tr>
                                <th>
                                    <Link to={`/add`}>
                                        <button type="button" className="btn btn-light btn-sm m-0">Create a new Campaign</button>
                                    </Link>
                                </th>
                                <th>Open (unsent) Campaigns</th>
                                <th>Target Segment</th>
                            </tr>
                        </thead>
                        <tbody>
                            {openCampaigns.map(campaign => {
                                const segment = segments.filter(segment => segment.id === campaign.segment_id);

                                return (
                                    <tr key={campaign.id}>
                                        <td>
                                            <Link to={`/edit/${campaign.id}`}>
                                                <button type="button" className="btn btn-light btn-sm m-0" onClick={() => editCampaign(campaign.id)}>Edit Details</button>
                                            </Link>
                                        </td>
                                        <td>{campaign.name}</td>
                                        <td>{segment[0].name}</td>
                                    </tr>
                                )
                            })}
                        </tbody>
                    </Table>
                </div>
            </Row>
        </Container>
    )
}
