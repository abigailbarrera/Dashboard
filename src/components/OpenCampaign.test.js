import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import { OpenCampaigns } from './OpenCampaigns';
 
describe('OpenCampaigns', () => {
  it('snapshot renders', () => {
    let tree = create(
        <BrowserRouter>
            <OpenCampaigns />
        </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders header', () => {
    const wrapper = shallow(<OpenCampaigns />);
    const tableHeader = "Open (unsent) Campaigns";

    expect(wrapper.contains(tableHeader)).toEqual(true);
  });
});
