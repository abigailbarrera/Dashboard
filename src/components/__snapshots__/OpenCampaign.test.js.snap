// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenCampaigns snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 mt-5"
    >
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              <a
                href="/add"
                onClick={[Function]}
              >
                <button
                  className="btn btn-light btn-sm m-0"
                  type="button"
                >
                  Create a new Campaign
                </button>
              </a>
            </th>
            <th>
              Open (unsent) Campaigns
            </th>
            <th>
              Target Segment
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <a
                href="/edit/4"
                onClick={[Function]}
              >
                <button
                  className="btn btn-light btn-sm m-0"
                  onClick={[Function]}
                  type="button"
                >
                  Edit Details
                </button>
              </a>
            </td>
            <td>
              Fall Decor Updates
            </td>
            <td>
              Decor Purchases
            </td>
          </tr>
          <tr>
            <td>
              <a
                href="/edit/5"
                onClick={[Function]}
              >
                <button
                  className="btn btn-light btn-sm m-0"
                  onClick={[Function]}
                  type="button"
                >
                  Edit Details
                </button>
              </a>
            </td>
            <td>
              Great Halloween Sale
            </td>
            <td>
              All Subscribers
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
